sources/packages/api/src/services/application/application.service.ts

```diff
@@ -638,4 +646,118 @@ export class ApplicationService extends RecordDataModelService<Application> {
       });
     return query.getOne();
   }
+
+  /**
+   * Creates a new Student Application overriding the current one
+   * in order to rollback the process and start the assessment
+   * all over again.
+   * @param locationId location id executing the COE rollback.
+   * @param applicationId application to be rolled back.
+   * @returns the overridden and the newly created application.
+   */
+  async overrideApplicationForCOE(
+    locationId: number,
+    applicationId: number,
+  ): Promise<ApplicationOverriddenResult> {
+    const appToOverride = await this.repo.findOne({
+      id: applicationId,
+      location: { id: locationId },
+    });
+
+    if (!appToOverride) {
+      throw new CustomNamedError(
+        "Student Application not found or the location does not have access to it",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (
+      appToOverride.applicationStatus !== ApplicationStatus.enrollment ||
+      appToOverride.coeStatus !== COEStatus.required
+    ) {
+      throw new CustomNamedError(
+        `Student Application is not in the expected status. The application must be in application status '${ApplicationStatus.enrollment}' and COE status '${COEStatus.required}' to be override.`,
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
+
+    // Change status of the current application being overwritten.
+    appToOverride.applicationStatus = ApplicationStatus.overwritten;
+    appToOverride.applicationStatusUpdatedOn = getUTCNow();
+
+    // Create a new application based in the previous one.
```

Comment:
Wording Suggestion: Create a new application record based off Overwritten record to rollback state of application

Reply:
Done